config/initializers/prometheus.rb

```diff
@@ -1,45 +1,60 @@
 # The below is heavily cribbed from ministryofjustice/laa-apply-for-criminal-legal-aid
 return unless ENV.fetch("ENABLE_PROMETHEUS_EXPORTER", "false") == "true"
 
+require 'prometheus_exporter/client'
 require "prometheus_exporter/server"
+require "prometheus_exporter/instrumentation"
+require "prometheus_exporter/middleware"
 
 DEFAULT_PREFIX = "ruby_".freeze
+DEFAULT_BIND_ADDRESS = "0.0.0.0".freeze
+DEFAULT_PORT = 9394
 
 # We are running puma in single process mode, so this is safe
 # If we move to multi process mode, we will have to run the
 # exporter process separately (`bundle exec prometheus_exporter`)
 def start_prometheus_server
   server = PrometheusExporter::Server::WebServer.new(
-    bind: "0.0.0.0", port: ENV.fetch("PROMETHEUS_EXPORTER_PORT", 9394).to_i,
+    bind: ENV.fetch("PROMETHEUS_EXPORTER_HOST", DEFAULT_BIND_ADDRESS),
+    port: ENV.fetch("PROMETHEUS_EXPORTER_PORT", DEFAULT_PORT).to_i,
     verbose: ENV.fetch("PROMETHEUS_EXPORTER_VERBOSE", "false") == "true"
   )
 
   server.start
-
-  true
+  server
 rescue Errno::EADDRINUSE
   warn "[PrometheusExporter] Server port already in use."
   false
 end
 
-return unless start_prometheus_server
+Rails.logger.info("[PrometheusExporter] Starting server....")
+server = start_prometheus_server
 
-require "prometheus_exporter/instrumentation"
-require "prometheus_exporter/middleware"
+Rails.logger.info "[PrometheusExporter] server #{JSON.parse(server.to_json).fetch_values('port', 'bind')}...."
+return unless server
+
+PrometheusExporter::Client.default = PrometheusExporter::LocalClient.new(collector: server.collector)
 
-Rails.logger.info "[PrometheusExporter] Initialising instrumentation middleware..."
+Rails.logger.info("[PrometheusExporter] Initialising standard instrumentation middleware...")
 
 # Metrics will be prefixed, for example `ruby_http_requests_total`
 PrometheusExporter::Metric::Base.default_prefix = DEFAULT_PREFIX
 
 # This reports stats per request like HTTP status and timings
+#
 Rails.application.middleware.unshift PrometheusExporter::Middleware
 
 # This reports basic process stats like RSS and GC info, type master
-# means it is instrumenting the master process
+# means it is instrumenting the master process.
+# see config/puma.rb  after_work_boot for more
+#
 PrometheusExporter::Instrumentation::Process.start(type: "master")
```

Comment:
do we still want to start this for the sidekiq server given we start a different one celow on the server?

Reply:
running on both does mean we could collect metrics per pod type (app or worker) but we can remove for now to avoid confusing aggregations between these worker/app pod stats for our future selves.

What do you think?

Reply:
have added this commit to remove https://github.com/ministryofjustice/laa-submit-crime-forms/pull/926/commits/dec16cb7af56a4b2f78f0bb0d15b177c2e98b06e